k9mail/src/main/java/com/fsck/k9/mailstore/CryptoResultAnnotation.java

```diff
@@ -72,8 +72,14 @@ public static CryptoResultAnnotation createOpenPgpCanceledAnnotation() {
         return new CryptoResultAnnotation(CryptoError.OPENPGP_UI_CANCELED, null, null, null, null, null);
     }
 
-    public static CryptoResultAnnotation createOpenPgpErrorAnnotation(OpenPgpError error) {
-        return new CryptoResultAnnotation(CryptoError.OPENPGP_API_RETURNED_ERROR, null, null, null, null, error);
+    public static CryptoResultAnnotation createOpenPgpSignatureErrorAnnotation(
+            OpenPgpError error, MimeBodyPart replacementData) {
+        return new CryptoResultAnnotation(
+                CryptoError.OPENPGP_SIGNED_API_ERROR, replacementData, null, null, null, error);
```

Comment:
On a general note: This is rather painful to read. A static factory method might make things look nicer.

```java
return CryptoResultAnnotation.errorWithReplacementData(CryptoError.OPENPGP_SIGNED_API_ERROR, error, replacementData);
```